config/config.go

```diff
@@ -0,0 +1,89 @@
+package config
+
+import (
+	"flag"
+	"github.com/caarlos0/env"
+	"github.com/northmule/gophermart/internal/app/services/logger"
+	"os"
+	"strings"
+)
+
+const DataBaseConnectionTimeOut = 10000
+const serverURLDefault = ":8081"
+const databaseURIDefault = "postgres://postgres:123@localhost:5456/gofermart?sslmode=disable"
+const accrualURLDefault = "http://localhost:8091"
+
+type GophermartConfig struct {
+	// Адрес сервера и порт
+	ServerURL string `env:"RUN_ADDRESS"`
+	// Внешняя система расчёта бонусов
+	AccrualURL string `env:"ACCRUAL_SYSTEM_ADDRESS"`
+	// Строка подключения к БД
+	DatabaseURI string `env:"DATABASE_URI"`
+}
+
+func NewGophermartConfig() (*GophermartConfig, error) {
+	instance := &GophermartConfig{}
+	err := instance.env()
+	if err != nil {
+		return nil, err
+	}
+
+	return instance, instance.flag()
```

Comment:
Элегантно, но не совсем хорошо читается. То, что `instance` обновит значения при вызове `instance.flag()` не совсем очевидно. Я бы сделал проще:
```suggestion
         if err := instance.flag(); err != nil {
                 return nil, err
          }
	return instance, nil
```